sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -391,9 +468,41 @@ static RunQueryRequest makeRequest(Query query, @Nullable String namespace) {
     }
 
     /**
+     * Returns a new {@link DatastoreV1.Read} that reads the results of the specified GQL query.
+     * See <a href="https://cloud.google.com/datastore/docs/reference/gql_reference">GQL Reference
+     * </a> to know more about GQL grammar.
+     *
+     * <p><b><i>Experimental</i></b>: Cloud Datastore does not a provide a clean way to translate
+     * a gql query string to {@link Query}, so we end up making a query to the service for
+     * translation but this may read the actual data. It needs more validation through production
```

Comment:
That is why I used `may`. Updated the comment to reflect that it is only a small amount.